apps/generic-issuance-client/src/helpers/userHooks.ts

```diff
@@ -0,0 +1,26 @@
+import { useStytch } from "@stytch/react";
+import { useEffect } from "react";
+import { SESSION_DURATION_MINUTES } from "../constants";
+
+/**
+ * If the current user session exists, return the current JWT and refresh
+ * the session on mount and before it expires. Otherwise, return undefined.
+ */
+export function useJWT(): string | undefined {
+  const { session } = useStytch();
+  useEffect(() => {
+    const refresh = (): void => {
+      if (session.getSync()) {
+        session.authenticate({
+          session_duration_minutes: SESSION_DURATION_MINUTES
+        });
+      }
+    };
+    // Refresh session on mount and halfway through the session.
+    refresh();
+    const interval = setTimeout(refresh, SESSION_DURATION_MINUTES / 2);
+    return () => clearTimeout(interval);
```

Comment:
something about this logic doesn't check out for me - shouldn't this be dependent on the amount of minutes *left* somehow?

Reply:
we already know what the number of minutes left is at the point when the `refresh()` scheduled with `setTimeout()` runs, since we set the session duration ourselves. one thing is that this should actually be a `setInterval()` rather than a `setTimeout()`, as we will need to keep scheduling `refresh()` to ensure it doesn't expire afterwards.